examples/cart/index.js

```diff
@@ -151,48 +165,59 @@ $(function() {
       $('.cart').removeClass('js-active');
     });
   }
-  closeCart();
 
   /* Add 'quantity' amount of product 'variant' to cart
   ============================================================ */
-  function addVariantToCart(product, variant, quantity) {
+  function addVariantToCart(variant, quantity) {
     openCart();
 
-    cart.addVariants({ variant: variant, quantity: quantity }).then(function () {
-      var $cartItemContainer = $('.cart-item-container');
-      var totalPrice = 0;
-
-      $cartItemContainer.empty();
-      var lineItemEmptyTemplate = $('#cart-item-template').html();
-      var $cartLineItems = cart.lineItems.map(function (lineItem, index) {
-        var $lineItemTemplate = $(lineItemEmptyTemplate);
-        var itemImage = lineItem.image.src;
-        $lineItemTemplate.find('.cart-item__img').css('background-image', 'url(' + itemImage + ')');
-        $lineItemTemplate.find('.cart-item__title').text(lineItem.title);
-        $lineItemTemplate.find('.cart-item__variant-title').text(lineItem.variant_title);
-        $lineItemTemplate.find('.cart-item__price').text(formatAsMoney(lineItem.line_price));
-        $lineItemTemplate.find('.cart-item__quantity').attr('value', lineItem.quantity);
-        $lineItemTemplate.find('.quantity-decrement').attr('data-variant-id', lineItem.variant_id);
-        $lineItemTemplate.find('.quantity-increment').attr('data-variant-id', lineItem.variant_id);
-
-        if (index === cart.lineItems.length - 1) {
-          $lineItemTemplate.addClass('js-hidden');
-        }
-        return $lineItemTemplate;
-      });
-      $cartItemContainer.append($cartLineItems);
-
-      setTimeout(function () {
-        $cartItemContainer.find('.js-hidden').removeClass('js-hidden');
-      }, 0)
-
-      $('.cart .pricing').text(formatAsMoney(cart.subtotal));
+    var itemExists;
+    if (cart.lineItems.length > 0) {
+      itemExists = cart.lineItems.filter(function (cartItem) {
```

Comment:
Instead of doing this filter, maybe check the length of `lineItems` and see if length grows. I think that's what's trigger your animation anyway, right? Appending to the list?